client/components/theme-upgrade-modal/index.tsx

```diff
@@ -178,28 +154,76 @@ export const ThemeUpgradeModal = ( {
 	};
 
 	const getStandardPurchaseModalData = (): UpgradeModalContent => {
-		const planPrice = premiumPlanProduct?.combined_cost_display;
-
-		return {
-			header: (
-				<h1 className="theme-upgrade-modal__heading">{ translate( 'Unlock this theme' ) }</h1>
-			),
-			text: (
-				<p>
-					{ translate(
-						'Get access to this theme, and a ton of other features, with a subscription to the %(premiumPlanName)s plan. It’s {{strong}}%(planPrice)s{{/strong}} a year, risk-free with a 14-day money-back guarantee.',
+		const getPlanText = ( planName: string, term: string, planPrice: string ) => {
+			switch ( term ) {
+				case 'three years':
+					return translate(
+						'Get access to this theme, and a ton of other features, with a subscription to the %(planName)s plan. It’s {{strong}}%(planPrice)s{{/strong}} per three years, risk-free with a 14-day money-back guarantee.',
 						{
 							components: {
 								strong: <strong />,
 							},
 							args: {
 								planPrice: planPrice || '',
-								premiumPlanName: premiumPlanName,
+								planName: planName,
 							},
 						}
-					) }
-				</p>
+					);
+				case 'two years':
+					return translate(
+						'Get access to this theme, and a ton of other features, with a subscription to the %(planName)s plan. It’s {{strong}}%(planPrice)s{{/strong}} per two years, risk-free with a 14-day money-back guarantee.',
+						{
+							components: {
+								strong: <strong />,
+							},
+							args: {
+								planPrice: planPrice || '',
+								planName: planName,
+							},
+						}
+					);
+				case 'month':
+					return translate(
+						'Get access to this theme, and a ton of other features, with a subscription to the %(planName)s plan. It’s {{strong}}%(planPrice)s{{/strong}} per month, risk-free with a 7-day money-back guarantee.',
+						{
+							components: {
+								strong: <strong />,
+							},
+							args: {
+								planPrice: planPrice || '',
+								planName: planName,
+							},
+						}
+					);
+				case 'year':
+				default:
+					return translate(
+						'Get access to this theme, and a ton of other features, with a subscription to the %(planName)s plan. It’s {{strong}}%(planPrice)s{{/strong}} annually, risk-free with a 14-day money-back guarantee.',
+						{
+							components: {
+								strong: <strong />,
+							},
+							args: {
+								planPrice: planPrice || '',
+								planName: planName,
+							},
+						}
+					);
+			}
+		};
+		const planPrice = requiredPlanProduct?.combined_cost_display;
+
+		const planText = getPlanText(
+			premiumPlanName as string,
```

Comment:
`premiumPlanName` comes from [`Plan.getTitle` which has a `TranslateResult` type](https://github.com/Automattic/wp-calypso//blob/66d66147dd4970440c6fc1767fffd6cf9d2ae5c9/packages/calypso-products/src/types.ts#L309), I'm pretty sure it could be typed as a string though

Reply:
Hmmm we need to be sure it's translated before casting (which I think it always is at this point).
`type ExistingReactNode = React.ReactElement | string | number;`

Reply:
I'm not sure that matters, it ought to always be a string, it'd be odd to return a ReactElement or a number from getTitle right?

When a translation resolves it triggers a re-render.